### PR TITLE
Override Jetpack connection data endpoint when Jetpack is not installed

### LINKED ISSUE #38979: Core Profiler - Jetpack Connection package conflicts with Android app flow
Reported pb22l9-27d-p2#comment-2286

### Description

The new core profiler conflicts with one of the Woo Android app's flows. Users are unable to install Jetpack from the app.

### Cause

The new core profiler includes Jetpack Connection package, which registers [/jetpack/v4/connection/data](https://href.li/?https://github.com/Automattic/jetpack-connection/blob/trunk/src/class-rest-connector.php#L133) REST API. The app uses the endpoint to fetch Jetpack status to determine whether it should install Jetpack.

When the endpoint returns 404 -> The app thinks Jetpack is not installed
When the endpoint returns a valid response -> The app thinks Jetpack is installed and tries to activate it.

### Solution

Since the endpoint is not used anywhere, we can safely override it when Jetpack is not installed and activated.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #38979 

This PR overrides `/jetpack/v4/connection/data` endpoint to return 404 when Jetpack is not installed to fix conflict with Android app Jetpack installation flow.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout the branch.
2. Try to access `http://YOUR-SITE/wp-json/jetpack/v4/connection/data` with wget `wget -U "Android" http://YOUR-SITE/wp-json/jetpack/v4/connection/data`
3. Confirm the endpoint returns 404 not found.
4. Install and activate Jetpack.
5. Try to access the endpoint again.
6. You should get a permission error.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement

#### Comment 

</details>

### CODE REVIEW

**Review by chihsuan — APPROVED** [review]
LGTM and tested well. 🚢 

> Try to access the endpoint again.
> You should get a permission error.

I've also tested this with a credential and got the 200 response.

File: plugins/woocommerce/src/Admin/API/OnboardingPlugins.php [hunk 1]
```diff
@@ -124,6 +125,29 @@ function ( $value ) {
...
 			)
 		);
+
+		/*
+		 * This is a temporary solution to override /jetpack/v4/connection/data endpoint
+		 * registered by Jetpack Connection when Jetpack is not installed.
+		 *
+		 * For more details, see https://github.com/woocommerce/woocommerce/issues/38979
+		 */
+		if ( Constants::get_constant( 'JETPACK__VERSION' ) === null ) {
```
Comment (adrianduffell) [inline plugins/woocommerce/src/Admin/API/OnboardingPlugins.php]: To reduce the chance of unknown side-effects, perhaps we could also add an [wp_is_mobile](https://developer.wordpress.org/reference/functions/wp_is_mobile/) check to limit the change to mobile devices only (or even better if we could limit it to requests from the android app).

**Review by adrianduffell — COMMENTED** [review]
@moon0326 What would be a good way to test the connection flow still works from the package?

I've left a nit suggestion about fixing it for mobile / android app requests only

File: plugins/woocommerce/src/Admin/API/OnboardingPlugins.php (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce/src/Admin/API/OnboardingPlugins.php]: That's a good suggestion! I'll make a change.

> What would be a good way to test the connection flow still works from the package?

There's no change for the existing flows. Just make sure to delete Jetpack before testing.

Comment (moon0326) [reply plugins/woocommerce/src/Admin/API/OnboardingPlugins.php]: Updated in https://github.com/woocommerce/woocommerce/pull/38987/commits/203722e4c3e6c89e922cc7fe4b79fb55095dc0a6

**Pull request merged**